kong/db/schema/typedefs.lua

```diff
@@ -376,6 +391,11 @@ typedefs.utf8_name = Schema.define {
   custom_validator = validate_utf8_name
 }
 
+typedefs.workspace_name = Schema.define {
+  type = "string",
+  unique = true,
+  custom_validator = validate_workspace_name
```

Comment:
```suggestion
  custom_validator = validate_workspace_name,
  not_one_of = constants.CORE_ENTITIES,
```

Should be sufficient. We don't have to re-implement all the logic in the `custom_validator`, they should both be executed with `not_one_of` checked first and `custom_validator` checked next.

Reply:
Good idea, @dndx, but it won't work as is, since `CORE_ENTITIES` behaves as an array and dictionary. What about the following?
```
diff --git a/kong/db/schema/entities/workspaces.lua b/kong/db/schema/entities/workspaces.lua
index 1709dc77..21c6b062 100644
--- a/kong/db/schema/entities/workspaces.lua
+++ b/kong/db/schema/entities/workspaces.lua
@@ -1,4 +1,5 @@
 local typedefs = require "kong.db.schema.typedefs"
+local constants = require "kong.constants"
 
 
 return {
@@ -10,7 +11,7 @@ return {
 
   fields = {
     { id          = typedefs.uuid },
-    { name        = typedefs.utf8_name { required = true } },
+    { name        = typedefs.utf8_name { required = true, not_one_of = { table.unpack(constants.CORE_ENTITIES) }, } },
     { comment     = { type = "string" } },
     { created_at  = typedefs.auto_timestamp_s },
     { meta        = { type = "record", fields = {} } },
```

This preserves the original typedef, only extending it with a validator.

Reply:
> tests fail with:
> 
> `kong/db/schema/entities/workspaces.lua:14:74: accessing undefined field unpack of global table`
> 
> while it passes locally.. Any idea?

That's interesting. Mind trying `unpack` (global) instead?